### PR TITLE
#1744 - Bulk Dataset Member Creation

### PR DESCRIPTION
**Pull request by JohnPTobe**
##### Checklist

- [x] `manage.py test` passes
- [x] tests are included
- [x] documentation is changed or added

### Affected app(s)
- dataset

### Description of change
Adding the ability to create multiple dataset members at once.

### CODE REVIEW

**Comment by JohnPTobe** [comment]
Sorry for the mess. I wanted to move the dataset code to the data app. It makes more sense to me to have all the data code in the same app.

File: scale/data/models.py [hunk 1]
```diff
@@ -365,26 +457,23 @@ class Meta(object):
...
-        file_ids = []
         datasetfiles = []
         for i in data.values.keys():
             v = data.values[i]
             if type(v) is FileValue:
-                file_ids.extend(v.file_ids)
                 for id in v.file_ids:
-                    exists = DataSetFile.objects.filter(dataset=dataset, scale_file_id=id).count()
-                    if exists:
+                    if id in existing_scale_ids:
```
Comment (emimaesmith) [inline scale/data/models.py]: I'm guessing this is to make sure we don't get duplicates?

Comment (JohnPTobe) [reply scale/data/models.py]: Yes, and looking at it I realized I forgot to append to that list as I go along so there aren't any duplicates in the files created from the data list.

File: scale/docs/rest/v6/data.rst [hunk 2]
```diff
@@ -490,27 +498,111 @@ Location http://.../v6/datasets/105/
...
+   
+Response: 200 Ok
+
+.. code-block:: javascript
 
+   [ <:ref:`Data JSON <rest_v6_data_data>`> ]
+   
 +-------------------------------------------------------------------------------------------------------------------------+
-| **Create Dataset Member*                                                                                                |
+| **Create Dataset Members*                                                                                               |
```
[the remark this answers is not in the document]
Comment (JohnPTobe) [reply scale/docs/rest/v6/data.rst]: I fixed the examples so there's a request with a data array that's not a dry run and a template/filter that is a dry run, with responses for both.

**Pull request merged**